### DISCUSSION TITLE
Videos from MPD source getting corrupted.

### DISCUSSION THREAD

**Discussion by DrErmani** [aclap-dev/video-downloadhelper#1534] [Help and Q&A]
### Your question

I'm trying to download videos that appear in the original format as MPD, when the download finishes they become defective and corrupted. All videos on the platform that are in this format have the same problem.

![imagen](https://github.com/user-attachments/assets/6944107b-dec1-406f-b508-f77b0f56040f)

![Corrupted Video](https://github.com/user-attachments/assets/22ec1776-a6f8-46ed-b53a-52346821947b)

### Addon & CoApp version and details

version: 9.1.0.44
target: mozilla
channel: stable
lang: es-ES
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"1ba0d5242d9e4cac"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:133.0) Gecko/20100101 Firefox/133.0
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "download_directory": "C:\\Users\\cdmg1\\OneDrive\\Ermani - Marcos\\Cursos\\Dr. Gabriel Almeida\\Mentoria de Terapias Injetáveis - Dezembro",
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "C:\\Users\\cdmg1\\OneDrive\\Ermani - Marcos\\Cursos\\Dr. Gabriel Almeida"
        }
    },
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 3,
    "successfull_dl": 7,
    "download_history": {
        "__serializer_tag": "map",
        "__serializer_value": [
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_1502686156871473"
                    },
                    {
                        "download_result": {
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "C:\\Users\\cdmg1\\OneDrive\\Ermani - Marcos\\Cursos\\Dr. Gabriel Almeida\\Mentoria de Terapias Injetáveis - Dezembro\\Discussao de Casos Clinicos em Terapias Injetaveis 19092024 Enco.mp4"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "Discussao de Casos Clinicos em Terapias Injetaveis 19092024 Enco.mp4"
                            },
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "C:\\Users\\cdmg1\\OneDrive\\Ermani - Marcos\\Cursos\\Dr. Gabriel Almeida\\Mentoria de Terapias Injetáveis - Dezembro"
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://injetaveis.drgabrielalmeida.com.br/aulas/discussao-de-casos-clinicos-em-terapias-injetaveis-19-09-2024-encontro-3/"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1733487929673
                        }
                    }
                ]
            }
        ]
    },
    "record_download_history": true
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

You need to search this forum for discussions of DRM.  There's plenty of them.

**Reply by GrampaWildWilly** [reply]
By the way, MPD is a red herring.  That's not what is causing your problem.  It's probably DRM.

**Reply by DrErmani** [reply]
What's is DRM? how can i fixed?

**Reply by Mattmjs** [reply]
@DrErmani , Digital Rights Management. It's used to prevent copying content online and your image is an example of that.
For your information as you're using Firefox, see this image that indicates what you're viewing is protected by DRM.

![DRM icon](https://github.com/user-attachments/assets/ba84f588-6504-45b9-a65a-f43b5ae423d5)
